cpp/src/arrow/flight/types.cc

```diff
@@ -430,6 +430,34 @@ std::string Location::scheme() const {
   return scheme;
 }
 
+std::string Location::path() const { return uri_->path(); }
+arrow::Result<std::vector<std::pair<std::string, std::string>>> Location::query_items()
+    const {
+  return uri_->query_items();
+}
+
+arrow::Result<std::vector<std::pair<std::string, std::string>>> Location::as_headers()
```

Comment:
This is far, far too specific to Flight SQL to belong as a method

Reply:
(As elsewhere, probably dropping these to keep things clean and consistent.)